x/hard/keeper/keeper.go

```diff
@@ -1,32 +1,33 @@
 package keeper
 
 import (
+	"encoding/json"
 	"time"
 
 	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/x/params/subspace"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
 
 	"github.com/kava-labs/kava/x/hard/types"
 )
 
 // Keeper keeper for the hard module
 type Keeper struct {
 	key             sdk.StoreKey
-	cdc             *codec.Codec
-	paramSubspace   subspace.Subspace
+	cdc             codec.Codec
+	paramSubspace   paramtypes.Subspace
 	accountKeeper   types.AccountKeeper
-	supplyKeeper    types.SupplyKeeper
+	bankKeeper      types.BankKeeper
 	stakingKeeper   types.StakingKeeper
```

Comment:
nit: looks like staking keeper isn't used anymore and could be removed. The method `BondDenom` isn't used.